src/ripple/app/misc/HashRouter.h

```diff
@@ -68,50 +68,42 @@ class HashRouter
         {
         }
 
-        std::set <PeerShortID> const& peekPeers () const
-        {
-            return peers_;
-        }
-
         void addPeer (PeerShortID peer)
         {
             if (peer != 0)
                 peers_.insert (peer);
         }
 
-        bool hasPeer (PeerShortID peer) const
-        {
-            return peers_.count (peer) > 0;
-        }
-
         int getFlags (void) const
         {
             return flags_;
         }
 
-        bool hasFlag (int mask) const
-        {
-            return (flags_ & mask) != 0;
-        }
-
         void setFlags (int flagsToSet)
         {
             flags_ |= flagsToSet;
         }
 
-        void clearFlag (int flagsToClear)
+        void swap (std::set <PeerShortID>& other)
         {
-            flags_ &= ~flagsToClear;
+            peers_.swap (other);
         }
 
-        void swapSet (std::set <PeerShortID>& other)
+        bool prepRelay (Stopwatch::time_point const& now,
```

Comment:
So, the first time we run through the function, we will always set `relayed_` to `now, ignore `holdTime` and return `true`. In subsequent runs, if the item was last "prepped for relay" less than `holdTime` second ago, we return `false` and do nothing, else we return `true`.

In other words, this function determines if the item should be relayed by checking whether it has been recently relayed. I'd rename it to something like `shouldRelay` or, the more abstract, `touch`.